src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -582,6 +585,15 @@ private static BaseListSyntax GetBaseListOpt(SingleTypeDeclaration decl)
             }
 
             return new Tuple<NamedTypeSymbol, ImmutableArray<NamedTypeSymbol>>(localBase, localInterfaces.ToImmutableAndFree());
+
+            void checkPrimaryConstructorBaseType(BaseTypeSyntax baseTypeSyntax, TypeSymbol baseType)
+            {
+                if (baseTypeSyntax is PrimaryConstructorBaseTypeSyntax primaryConstructorBaseType &&
+                    (!IsRecord || TypeKind == TypeKind.Struct || baseType.TypeKind == TypeKind.Interface || ((RecordDeclarationSyntax)decl.SyntaxReference.GetSyntax()).ParameterList is null))
```

Comment:
>TypeKind == TypeKind.Struct || baseType.TypeKind == TypeKind.Interface  [](http://example.com/codeflow?start=34&length=71)

```TypeKind != TypeKind.Class```? Only base class can have arguments.

Reply:
For the first part: `!= Class` and `== Struct` are equivalent since we're dealing with a record (`IsRecord` check earlier on the line).
For the second part, the reason I checked for interface is because the other scenarios all already have errors (see below).

Reply:
> For the first part: != Class and == Struct are equivalent since we're dealing with a record (IsRecord check earlier on the line).
For the second part, the reason I checked for interface is because the other scenarios all already have errors (see below). 

It always more robust to not depend on other conditions and checking for what is required for the "good" state rather than trying to cover other possible "bad" states. The arguments are allowed for classes on base classes. I believe that the condition should be coded in these terms, rather than trying to cover what else might be there based on some previos conditions.